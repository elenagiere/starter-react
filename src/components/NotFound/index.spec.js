import React from 'react';
import { shallow } from 'enzyme';
import NotFoundPage from './index';

describe('<NotFoundPage />', () => {
  it('should match SnapShot', () => {
    const container = shallow(<NotFoundPage />);
    expect(container).toMatchSnapshot();
  });

  it('should have a header called \'404 Not Found\'', () => {
    const wrapper = shallow(<NotFoundPage />);
    const actual = wrapper.find('.md-panel__title').text();
    const expected = '404 Not Found';

    expect(actual).toEqual(expected);
  });

  it('should have a message with \'The page you requested cannot be found.\'', () => {
    const wrapper = shallow(<NotFoundPage />);
    const actual = wrapper.find('.md-panel__message').text();
    const expected = 'The page you requested cannot be found.';

    expect(actual).toEqual(expected);
  });

  it('should link to an home', () => {
    const wrapper = shallow(<NotFoundPage />);
    const actual = wrapper.findWhere(n => n.prop('to') === '/').length;
    const expected = 1;

    expect(actual).toEqual(expected);
  });

  it('should have footer content', () => {
    const wrapper = shallow(<NotFoundPage />);
    const actual = wrapper.find('.footer__copyright').text();
    const expected = 'Â© 2018 Cisco and/or affiliates. All rights reserved.';

    expect(actual).toEqual(expected);
  });
});
