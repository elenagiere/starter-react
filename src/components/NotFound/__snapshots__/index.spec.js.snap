// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> should match SnapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFoundPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="md-panel__main"
>
          <i
                    className="md-panel__image icon icon-warning_100"
          />
          <div
                    className="md-panel__title"
          >
                    404 Not Found
          </div>
          <div
                    className="md-panel__message"
          >
                    The page you requested cannot be found.
          </div>
          <div
                    className="md-panel__cta"
          >
                    <Link
                              className="md-button md-button--blue"
                              replace={false}
                              to="/"
                    >
                              Go back to homepage
                    </Link>
          </div>
          <div
                    className="md-panel__secondary-action"
          >
                    <a
                              href="#"
                    >
                              Contact Support
                    </a>
          </div>
</div>,
        <div
          className="md-panel__footer"
>
          <div
                    className="footer__logo"
          >
                    <i
                              className="icon icon-cisco-logo"
                    />
          </div>
          <div
                    className="footer__copyright"
          >
                    © 2018 Cisco and/or affiliates. All rights reserved.
          </div>
</div>,
      ],
      "className": "md-panel md-panel--message md-panel--full",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="md-panel__image icon icon-warning_100"
/>,
            <div
              className="md-panel__title"
>
              404 Not Found
</div>,
            <div
              className="md-panel__message"
>
              The page you requested cannot be found.
</div>,
            <div
              className="md-panel__cta"
>
              <Link
                            className="md-button md-button--blue"
                            replace={false}
                            to="/"
              >
                            Go back to homepage
              </Link>
</div>,
            <div
              className="md-panel__secondary-action"
>
              <a
                            href="#"
              >
                            Contact Support
              </a>
</div>,
          ],
          "className": "md-panel__main",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "md-panel__image icon icon-warning_100",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "404 Not Found",
              "className": "md-panel__title",
            },
            "ref": null,
            "rendered": "404 Not Found",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "The page you requested cannot be found.",
              "className": "md-panel__message",
            },
            "ref": null,
            "rendered": "The page you requested cannot be found.",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="md-button md-button--blue"
                replace={false}
                to="/"
>
                Go back to homepage
</Link>,
              "className": "md-panel__cta",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Go back to homepage",
                "className": "md-button md-button--blue",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Go back to homepage",
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="#"
>
                Contact Support
</a>,
              "className": "md-panel__secondary-action",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Contact Support",
                "href": "#",
              },
              "ref": null,
              "rendered": "Contact Support",
              "type": "a",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="footer__logo"
>
              <i
                            className="icon icon-cisco-logo"
              />
</div>,
            <div
              className="footer__copyright"
>
              © 2018 Cisco and/or affiliates. All rights reserved.
</div>,
          ],
          "className": "md-panel__footer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="icon icon-cisco-logo"
/>,
              "className": "footer__logo",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "icon icon-cisco-logo",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "© 2018 Cisco and/or affiliates. All rights reserved.",
              "className": "footer__copyright",
            },
            "ref": null,
            "rendered": "© 2018 Cisco and/or affiliates. All rights reserved.",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="md-panel__main"
>
            <i
                        className="md-panel__image icon icon-warning_100"
            />
            <div
                        className="md-panel__title"
            >
                        404 Not Found
            </div>
            <div
                        className="md-panel__message"
            >
                        The page you requested cannot be found.
            </div>
            <div
                        className="md-panel__cta"
            >
                        <Link
                                    className="md-button md-button--blue"
                                    replace={false}
                                    to="/"
                        >
                                    Go back to homepage
                        </Link>
            </div>
            <div
                        className="md-panel__secondary-action"
            >
                        <a
                                    href="#"
                        >
                                    Contact Support
                        </a>
            </div>
</div>,
          <div
            className="md-panel__footer"
>
            <div
                        className="footer__logo"
            >
                        <i
                                    className="icon icon-cisco-logo"
                        />
            </div>
            <div
                        className="footer__copyright"
            >
                        © 2018 Cisco and/or affiliates. All rights reserved.
            </div>
</div>,
        ],
        "className": "md-panel md-panel--message md-panel--full",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="md-panel__image icon icon-warning_100"
/>,
              <div
                className="md-panel__title"
>
                404 Not Found
</div>,
              <div
                className="md-panel__message"
>
                The page you requested cannot be found.
</div>,
              <div
                className="md-panel__cta"
>
                <Link
                                className="md-button md-button--blue"
                                replace={false}
                                to="/"
                >
                                Go back to homepage
                </Link>
</div>,
              <div
                className="md-panel__secondary-action"
>
                <a
                                href="#"
                >
                                Contact Support
                </a>
</div>,
            ],
            "className": "md-panel__main",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "md-panel__image icon icon-warning_100",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "404 Not Found",
                "className": "md-panel__title",
              },
              "ref": null,
              "rendered": "404 Not Found",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "The page you requested cannot be found.",
                "className": "md-panel__message",
              },
              "ref": null,
              "rendered": "The page you requested cannot be found.",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="md-button md-button--blue"
                  replace={false}
                  to="/"
>
                  Go back to homepage
</Link>,
                "className": "md-panel__cta",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Go back to homepage",
                  "className": "md-button md-button--blue",
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Go back to homepage",
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="#"
>
                  Contact Support
</a>,
                "className": "md-panel__secondary-action",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Contact Support",
                  "href": "#",
                },
                "ref": null,
                "rendered": "Contact Support",
                "type": "a",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="footer__logo"
>
                <i
                                className="icon icon-cisco-logo"
                />
</div>,
              <div
                className="footer__copyright"
>
                © 2018 Cisco and/or affiliates. All rights reserved.
</div>,
            ],
            "className": "md-panel__footer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="icon icon-cisco-logo"
/>,
                "className": "footer__logo",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "icon icon-cisco-logo",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "© 2018 Cisco and/or affiliates. All rights reserved.",
                "className": "footer__copyright",
              },
              "ref": null,
              "rendered": "© 2018 Cisco and/or affiliates. All rights reserved.",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
